src/main/java/org/biouno/unochoice/util/Utils.java

```diff
@@ -243,11 +240,12 @@ public static boolean isDisabled(@Nullable Object obj) {
      * @param project the project to search for this parameter definition.
      * @return {@code true} if the project contains this parameter definition.
      */
-    private static boolean isParameterDefinitionOf(@Nonnull String parameterUUID, @Nonnull Project<?, ?> project) {
+    private static boolean isParameterDefinitionOf(@Nonnull String parameterUUID, @Nonnull Job<?, ?> project) {
         List<ParameterDefinition> parameterDefinitions = new ArrayList<>(getProjectParameterDefinitions(project));
-        for (List<ParameterDefinition> params : getBuildWrapperParameterDefinitions(project).values()) {
+        // TODO: Update or get rid of getBuildWrapperParameterDefinitions
```

Comment:
@kinow Dear maintainers, please help me with this part of the plugin. I'm unsure how we use `getBuildWrapperParameterDefinition` and how we test that it works.
Do we still need it with all these introspections with BuildWrapper and all?

Reply:
This `isParameterDefinitionOf` method is called as a fallback when we cannot find the project of the parameter.

We navigate all the projects in Jenkins, trying to find one that contains the parameter UUID. It is highly inefficient, but was the last resort after getting so many bugs from users with custom plug-ins that were modifying the structure of project <-> parameters (i.e. inserting extra nodes in-between these two).

It should work with pipeline jobs, no?